proposals/0000-migrate-to-androidx-alertdialog.md

```diff
@@ -0,0 +1,86 @@
+
+Migrating to `androidx.appcompat.app.AlertDialog` would mitigate most of the aforementioned issues as it implicitly enforces using an `AppCompat` theme and provides compatible dark mode support out of the box without having to ask developers to modify their setup.
+
+## Detailed design
+
+There are two primary changes to make: in `AlertFragment` and `DevSupportManagerBase`.
+
+The `AlertFragment` will have to be modified to use the `androidx.appcompat.app.AlertDialog` builder instead of `android.app.AlertDialog`'s builder. Since the class inherits from `DialogFragment`, it only needs to yield a `Dialog` type.
+
+Similarly, the `DevSupportManagerBase` will also have to make this change from `android.app.AlertDialog` to `androidx.appcompat.app.AlertDialog`.
+
+To support activities that do not have AppCompat support, we will need to perform runtime checks against the current activity. This is necessary to ensure nothing crashes, as `AppCompatDelegateImpl` throws an `IllegalStateException` if the current activity does not have an AppCompat theme.
+
+The check would look something like this:
+
+```kotlin
+fun isAppCompatTheme(activityContext: Context): Boolean {
+  val typedArray = activityContext.obtainStyledAttributes(R.styleable.AppCompatTheme)
+  return typedArray.hasValue(R.styleable.AppCompatTheme_windowActionBar)
+}
+```
+
+We would rework the static `createDialog` function that `AlertFragment` exposes, performing this check up front and logging warnings to users that fail the check:
+
+```kotlin
+fun createDialog(
+  activityContext: Context,
+  arguments: Bundle,
+  fragment: DialogInterface.OnClickListener ,
+): Dialog {
+  return if (isAppCompatTheme()) {
+    // return androidx.appcompat.app.AlertDialog result
+  } else {
+    // Warning logs here
+    // return android.app.AlertDialog result
+  }
+}
+```
+
+And in `DevSupportManagerBase` we would make the same check with logging. To ensure that users perform this upgrade right away, we should also make this check and perform logging whenever a ReactContext is created from this module.
+
+## Drawbacks
+
+There are a few reasons why we would not want to do this:
+- This deprecates support for non-AppCompat activities. We are explicitly choosing to drop support for the `android.app.AlertDialog` and its theming quirks
+- Theming is now applied via `alertDialogTheme`, not `android:alertDialogTheme`. This is a breaking change and needs to be validated by users who upgrade to the version that contains this change.
+
+## Alternatives
+
+I considered dropping support for `android.app.AlertDialog` altogether, primarily because `ReactActivity` already subclasses `AppCompatActivity`. However, this would negatively impact brownfield developers.
+
+The only other alternative is to leave this component as-is.
+
+## Adoption strategy
```

Comment:
Q: What's the timeline for this to be merged? (i.e. in which RN version)

Reply:
I've made a change in d16e750 that reflects the versioning strategy, but I believe the next minor release number (i.e. 0.75.0, not 0.74.3) would be best

Reply:
Yup that makes sense 👍